https://github.com/mlflow/mlflow/actions/runs/3376334447/jobs/5604026210#step:5:633

```
FAILED tests/test_runs.py::test_list_run - mlflow.exceptions.MlflowException: Both 'run_name' argument and 'mlflow.runName' tag are specified. Remove 'mlflow.runName' tag.
```

We currently do not allow specifying both `run_name` and `mlflow.runName` tag. We need to relax this constraint.

--- a/mlflow/store/tracking/file_store.py
+++ b/mlflow/store/tracking/file_store.py
@@ -579,10 +579,10 @@ class FileStore(AbstractStore):
             )
         tags = tags or []
         run_name_tag = _get_run_name_from_tags(tags)
-        if run_name and run_name_tag:
+        if run_name and run_name_tag and run_name != run_name_tag:
             raise MlflowException(
-                "Both 'run_name' argument and 'mlflow.runName' tag are specified. "
-                "Remove 'mlflow.runName' tag.",
+                "Both 'run_name' argument and 'mlflow.runName' tag are specified, but with "
+                f"different values (run_name='{run_name}', mlflow.runName='{run_name_tag}').",
                 INVALID_PARAMETER_VALUE,
             )
         run_name = run_name or run_name_tag or _generate_random_name()

--- a/mlflow/store/tracking/sqlalchemy_store.py
+++ b/mlflow/store/tracking/sqlalchemy_store.py
@@ -470,10 +470,10 @@ class SqlAlchemyStore(AbstractStore):
             )
             tags = tags or []
             run_name_tag = _get_run_name_from_tags(tags)
-            if run_name and run_name_tag:
+            if run_name and run_name_tag and (run_name != run_name_tag):
                 raise MlflowException(
-                    "Both 'run_name' argument and 'mlflow.runName' tag are specified. "
-                    "Remove 'mlflow.runName' tag.",
+                    "Both 'run_name' argument and 'mlflow.runName' tag are specified, but with "
+                    f"different values (run_name='{run_name}', mlflow.runName='{run_name_tag}').",
                     INVALID_PARAMETER_VALUE,
                 )
             run_name = run_name or run_name_tag or _generate_random_name()

--- a/tests/store/tracking/test_file_store.py
+++ b/tests/store/tracking/test_file_store.py
@@ -851,13 +851,16 @@ class TestFileStore(unittest.TestCase, AbstractStoreTest):
 
         with pytest.raises(
             MlflowException,
-            match=re.escape("Both 'run_name' argument and 'mlflow.runName' tag are specified."),
+            match=re.escape(
+                "Both 'run_name' argument and 'mlflow.runName' tag are specified, but with "
+                "different values (run_name='my name', mlflow.runName='test')."
+            ),
         ):
             fs.create_run(
                 experiment_id=FileStore.DEFAULT_EXPERIMENT_ID,
                 user_id="user",
                 start_time=0,
-                run_name="test",
+                run_name="my name",
                 tags=[RunTag(MLFLOW_RUN_NAME, "test")],
             )
 

--- a/tests/store/tracking/test_rest_store.py
+++ b/tests/store/tracking/test_rest_store.py
@@ -148,6 +148,7 @@ class TestRestStore:
 
         user_name = "mock user"
         source_name = "rest test"
+        run_name = "my name"
 
         source_name_patch = mock.patch(
             "mlflow.tracking.context.default_context._get_source_name", return_value=source_name
@@ -163,17 +164,18 @@ class TestRestStore:
         ), mock.patch(
             "time.time", return_value=13579
         ), source_name_patch, source_type_patch:
-            with mlflow.start_run(experiment_id="43", run_name="my name"):
+            with mlflow.start_run(experiment_id="43", run_name=run_name):
                 cr_body = message_to_json(
                     CreateRun(
                         experiment_id="43",
                         user_id=user_name,
-                        run_name="my name",
+                        run_name=run_name,
                         start_time=13579000,
                         tags=[
                             ProtoRunTag(key="mlflow.source.name", value=source_name),
                             ProtoRunTag(key="mlflow.source.type", value="LOCAL"),
                             ProtoRunTag(key="mlflow.user", value=user_name),
+                            ProtoRunTag(key="mlflow.runName", value=run_name),
                         ],
                     )
                 )

--- a/tests/store/tracking/test_sqlalchemy_store.py
+++ b/tests/store/tracking/test_sqlalchemy_store.py
@@ -853,14 +853,17 @@ class TestSqlAlchemyStore(unittest.TestCase, AbstractStoreTest):
 
         with pytest.raises(
             MlflowException,
-            match=re.escape("Both 'run_name' argument and 'mlflow.runName' tag are specified."),
+            match=re.escape(
+                "Both 'run_name' argument and 'mlflow.runName' tag are specified, but with "
+                "different values (run_name='test', mlflow.runName='test_2').",
+            ),
         ):
             self.store.create_run(
                 experiment_id=experiment_id,
                 user_id="user",
                 start_time=0,
                 run_name="test",
-                tags=[RunTag(mlflow_tags.MLFLOW_RUN_NAME, "test")],
+                tags=[RunTag(mlflow_tags.MLFLOW_RUN_NAME, "test_2")],
             )
 
     def test_get_run_with_name(self):

